src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationOptions.cs

```diff
@@ -154,25 +156,34 @@ public AzureAppConfigurationOptions UseFeatureFlags(Action<FeatureFlagOptions> c
                     string.Format(ErrorMessages.CacheExpirationTimeTooShort, MinimumFeatureFlagsCacheExpirationInterval.TotalMilliseconds));
             }
 
-            if (!(_kvSelectors.Any(selector => selector.KeyFilter.StartsWith(FeatureManagementConstants.FeatureFlagMarker) && selector.LabelFilter.Equals(options.Label))))
+            string fullPrefix = FeatureManagementConstants.FeatureFlagMarker + options.Prefix;
+
+            if (!(_kvSelectors.Any(selector => selector.KeyFilter.StartsWith(fullPrefix) && selector.LabelFilter.NormalizeNull() == options.Label.NormalizeNull())))
```

Comment:
>NormalizeNull [](start = 113, length = 13)

Do you see a scenario where `selector.LabelFilter` could be `null`?

Reply:
`selector.LabelFilter` should always be `LabelFilter.Null` but `options.Label` could be either `null` or `LabelFilter.Null`.

Reply:
Right! Could we get rid of the `NormalizeNull` call on `selector.LabelFilter`?

Reply:
I thought that we would end up with duplicate selectors if we don't `NormalizeNull` here. But after looking into `Select` logic more closely, I think we can remove the `NormalizeNull` call for both `selector.LabelFilter` and `options.Label`. `Select` method will take care of `null` and `LabelFilter.Null` appropriately.